ingestion/src/metadata/utils/secrets_manager.py

```diff
@@ -36,24 +46,40 @@
     "ServiceConnectionType", ServiceConnection.__fields__["__root__"].type_
 )
 
+AuthProviderClientType = NewType(
+    "AuthProviderClientType", OpenMetadataConnection.__fields__["securityConfig"].type_
+)
+
+AUTH_PROVIDER_MAPPING: Dict[AuthProvider, AuthProviderClientType] = {
+    AuthProvider.google: googleSSOClientConfig.GoogleSSOClientConfig,
+    AuthProvider.okta: oktaSSOClientConfig.OktaSSOClientConfig,
+    AuthProvider.auth0: auth0SSOClientConfig.Auth0SSOClientConfig,
+    AuthProvider.azure: azureSSOClientConfig.AzureSSOClientConfig,
+    AuthProvider.custom_oidc: customOidcSSOClientConfig.CustomOIDCSSOClientConfig,
+    AuthProvider.openmetadata: openMetadataJWTClientConfig.OpenMetadataJWTClientConfig,
+}
+
 
 class SecretsManager(metaclass=Singleton):
```

Comment:
nit - it would be interesting to add some docstring in the class and function to help add new implementations of this.

Reply:
also when we add the linting, it will ask for the docstring there